### PR TITLE
Re former setup

### PR DESCRIPTION
**Pull request by sanadwj**
## what this PR for:
- Generate the rais app
- Create user model and validations
- Create a user controller and it's methods 
- Add the HTML forms, form_tag, form_for
- Add edit method

### CODE REVIEW

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<%#
+<form accept-charset="UTF-8" action="/users" method="post" >
```
Comment (mosesogwo) [inline app/views/users/new.html.erb:2]: Use the rails _users\_path_ helper for the action option here.

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  <input type="hidden" name="authenticity_token" value="<%= form_authenticity_token
```
Comment (mosesogwo) [inline app/views/users/new.html.erb:3]: Where is the closing ERB tag for authenticity token here?

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  <%= text_field_tag(:password) 
+  <%= submit_tag("Submit") 
+<% end 
+%>
+
+<%= form_for @user, {class: "nifty_form"} do |f| %>
+  <%= f.label :user_name, "Name" %>
+  <%= f.text_field :user_name, placeholder:"Name" %> 
+  <%= f.label :email, "E-mail" %>
+  <%= f.text_field :email, placeholder:"Email" %>
```
Comment (mosesogwo) [inline app/views/users/new.html.erb:30]: Use **f.email_field** instead of f.text_field here

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hello :wave: Nice work on this :+1:**
Please see my comments on your code below and effect the required changes.

Here are some other **optional improvements** you could make
-  create a partial for user form and reuse in edit and new view
- on your form, you could display a list of the error messages if they exist (with  #errors and full_messages usage) 
- validate the email format by using either a regular expression or a built in URI method.

_**Kindly make these changes and submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested:exclamation:

**Comment by sanadwj** [comment]
@mosesogwo, Thanks for the review.
We edit all the changes

File: app/views/users/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<%#
+<form accept-charset="UTF-8" action="users_path" method="post" >
```
Comment (elmejdki) [inline app/views/users/_form.html.erb:2]: - user_path method goes without quotes into action.

File: app/views/users/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  <input type="hidden" name="authenticity_token" value="<%= form_authenticity_token >
```
Comment (elmejdki) [inline app/views/users/_form.html.erb:3]: you are messing the ERB closing tag in from_authenticity_token as mentioned by the previous TSE.

File: app/views/users/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  <input type="text" id="email" name="user[email]"><br><br>
+  <label for="password">Password:</label>
+  <input type="password" id="password" name="user[password]"><br><br>
+  <input type="submit" value="Submit">
+</form>
+%> 
+
+<% 
+=begin %>
+<%= form_tag("/users", method: "post") do %>
```
Comment (elmejdki) [inline app/views/users/_form.html.erb:16]: use also users_path in the form_tag as in here: 
` form_tag(users_path) do`

**Review by elmejdki — CHANGES_REQUESTED** [review]
# status: require changes :recycle: 
## Well done, guys :+1: :100: :clap: you are almost there just some couple changes :hammer: :wrench: and your project will be approved :checkered_flag: .

I have left some comments below please check them.

## Happy Coding :tophat:

**Review by BrittanyBlake — APPROVED** [review]
### Status: Approved ✅ 
****
### Hi Team! 👋 
### Great job implementing the requested changes!! 🎖️ It's a 👍 from me! Go ahead and merge this pull request and keep up the good work 😄
****

 Good Luck and Happy Coding! 🍀 👩🏾‍💻

**Pull request merged**